### PR TITLE
Pytwin cleanup temporary directory at python process exit

### PR DESCRIPTION
**Pull request by lboucin**
- This PR restore the previous behavior regarding pytwin temp working directory which was to cleanup this directory as some point in the pytwin execution. Legacy behavior was to cleanup the `%TEMP%/pytwin` folder at pytwin import. With PR103 and support for multiprocess execution this cleanup was removed. As a consequence, the pytwin temporary folder can become very big.
- With this PR, the pytwin settings module cleanup its process dependent temporary directory at python process exit. This makes the legacy cleanup behavior is restored while keeping multiprocess execution enabled.
- Unit tests cleanup helpers are modified to handle this new mechanism.
- Python contexts are also added to the `tbrom` module read/write files helpers.

### CODE REVIEW

**Review by lboucin — COMMENTED** [review]
Changes self reviewed. LGTM

File: src/ansys/pytwin/settings.py [hunk 1]
```diff
@@ -447,3 +465,11 @@ def modify_logging(new_option: PyTwinLogOption, new_level: PyTwinLogLevel):
 
 
 PYTWIN_SETTINGS = _PyTwinSettings()  # This instance is here to launch default settings initialization.
+
+
+@atexit.register
```
Comment (lboucin) [inline src/ansys/pytwin/settings.py:470]: @chrpetre This is the main modification of the business logic: `pytwin.settings` will remove its temporary working directory at python process exit, without modifying temporary working directory of any other python process having imported pytwin. This allows to have a cleanup of the pytwin temporary directory (it is kept empty at the end of the unit tests execution, as it was before PR103). Users that will want to keep pytwin working directory alive at the end of the python process will have to use the `modify_pytwin_working_directory` helper.

File: tests/test_settings.py [hunk 2]
```diff
@@ -325,5 +325,23 @@ def test_modify_working_after_logging_console(self):
         assert len(logger.handlers) == 1
         assert log_file is None
 
+    def test_multiprocess_execution_pytwin_cleanup_is_safe(self):
```
Comment (lboucin) [inline tests/test_settings.py:328]: @chrpetre @EDCarman This unit test checks the new business logic.

File: src/ansys/pytwin/settings.py (hunk 1 above)
Comment (chrpetre) [reply src/ansys/pytwin/settings.py:470]: thanks @lboucin that looks good to me. Just a couple of questions/comments 
- where is the new logic (default and how to change the default) explained/documented right now ? (e.g. modify_pytwin_working_directory could perhaps have some comments about it for users interested in keeping the temp directory alive)
- have you tested to change the working directory and verify it keeps well the files as expected in that case ?

Comment (lboucin) [reply src/ansys/pytwin/settings.py:478]: Hi @chrpetre, thanks for your suggestions:

- I've documented default pytwin temporary directory in the doc string (see [here](https://github.com/ansys/pytwin/blob/db75cee89fea3bb1c8fceb4259108110a140c6bd/src/ansys/pytwin/settings.py#L142)), as well as added a new option to not delete temp directory at exit. This behavior is unit tested [here](https://github.com/ansys/pytwin/blob/db75cee89fea3bb1c8fceb4259108110a140c6bd/tests/test_settings.py#L344C15-L344C15).
- Yes, the behavior is unit tested [here](https://github.com/ansys/pytwin/blob/db75cee89fea3bb1c8fceb4259108110a140c6bd/tests/evaluate/test_twin_model.py#L393C17-L393C17).

**Review by chrpetre — APPROVED** [review]
thanks Lucas

**Pull request merged**